reports/cucumber.api.md

```diff
@@ -0,0 +1,144 @@
+## API Report File for "@cucumber/cucumber"
```

Comment:
What is the purpose of that file?

Reply:
It's basically the source of truth for API Extractor. When it runs, it generates a fresh version of this in a staging file, then compares to this and checks if anything changed. So:

- If you unwittingly make a change to the API surface, it fails in CI
- When you run `npm run docs:local`, it overwrites this, and then the diff on the PR gives a nice summary of changes for the reviewer.

Reply:
Nice! Thanks for the explanation